core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -1453,6 +1642,11 @@ private[history] object LogType extends Enumeration {
   val DriverLogs, EventLogs = Value
 }
 
+private[history] object LogXAttrStatus extends Enumeration {
+  val XATTR_DISABLED, XATTR_ENABLED, APP_STARTED, APP_ENV_UPDATED, APP_END = Value
```

Comment:
Add documentation on what each of these states is capturing (in the lifecycle of a spark application)